### PR TITLE
feature 718 PCPCombine multiple fields

### LINKED ISSUE #718: Enhance PCPCombine wrapper to be able to process multiple fields in one command
This has been discussed before, but we discovered an issue did not exist when handling Mallory's MET Help ticket 97764. 

You should be able to process multiple fields at once in derive mode (in 1 command) and process multiple fields by looping through in other modes. I believe this can be achieved in some modes by using [FCST/OBS]\_VAR<n>\_* variables, but the logic should be updated to use the recommended variables, i.e. [FCST/OBS]\_PCP_COMBINE_INPUT\_[NAMES/LEVELS/ACCUMS] should be [FCST/OBS]\_PCP\_COMBINE\_VAR\<n>\_INPUT_[NAMES/LEVELS/ACCUMS]

## Describe the Enhancement ##
*Provide a description of the enhancement request here.*

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2791541 

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [x] Complete the development and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [x] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Added unit tests to test existing MET Tool Wrapper use cases work as intended, then added 2 tests to ensure that setting the new config variables create the desired command in -derive mode.
I reached out to a few people from NOAA (@malloryprow and @LoganDawson-NOAA ) who requested this feature to test if the new variables achieve what they need. Mallory already has a work-around in place using the "user-defined command" method of running PCPCombine wrapper.

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review code changes and automated test logs. Optionally test that a new field can be added to a use case. Optionally wait until other users (listed above) can confirm that the changes achieve what they need.

- [X] Do these changes include sufficient documentation and testing updates? **[Yes]**
Added new unit tests to ensure new functionality works as intended. Added new items to glossary and python wrappers pages of User's Guide

- [X] Will this PR result in changes to the test suite? **[Yes]**</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Users_Guide/glossary.rst [hunk 1]
```diff
@@ -5614,3 +5614,30 @@ METplus Configuration Glossary
      Specify the value for 'mask.grid' in the MET configuration file for GridStat.
 
      | *Used by:*  GridStat
+
+   FCST_PCP_COMBINE_EXTRA_NAMES
+     Specify a list of any additional field to add to the command. The number of items in this list should match :term:`FCST_PCP_COMBINE_EXTRA_LEVELS`. A corresponding variable exists for observation data called :term:`OBS_PCP_COMBINE_EXTRA_NAMES`. Example:
```
Comment (JohnHalleyGotway) [inline docs/Users_Guide/glossary.rst]:
```suggestion
     Specify a list of any additional fields to add to the command. The number of items in this list should match :term:`FCST_PCP_COMBINE_EXTRA_LEVELS`. A corresponding variable exists for observation data called :term:`OBS_PCP_COMBINE_EXTRA_NAMES`. Example:
```

**Review by JohnHalleyGotway — APPROVED** [review]
The code changes and documentation updates look great to me. There was one minor typo for which I made a suggested change.

I approve of these code and documentation changes. Prior to merging these changes in, recommend that you:
- Wait for the automated tests to complete
- Wait for @malloryprow or @LoganDawson-NOAA to confirm that this enhancement makes sense and meets their needs.

**Comment by georgemccabe** [comment]
Note that one of the use case tests failed (in climate) due to a change in the MET develop branch. The use case runs successfully, but the output data are named differently than the previous run. @JohnHalleyGotway is currently investigating.

**Comment by malloryprow** [comment]
I only see the option to "Merge pull request", but nothing to just approve it.

**Comment by georgemccabe** [comment]
@malloryprow click on the "Files changed" tab. You should see a button on the right that says "Review changes."
PR-approve

**Pull request merged**